js/notebook/src/plot/plotScope.js

```diff
@@ -1714,7 +1714,28 @@ define([
       }else{
         // scale only
         var level = self.zoomLevel;
-        if (my <= plotUtils.safeHeight(self.jqsvg) - self.layout.bottomLayoutMargin) {
+        var autoZoomSuccess = false;
+        if (my <= plotUtils.safeHeight(self.jqsvg) - self.layout.bottomLayoutMargin
+          && mx >= self.layout.leftLayoutMargin) {
+            // Zooming in the middle of the chart, autoscale Y
+            var data = this.stdmodel.data;
+            if (data.map(d => d.getRange ? true : false).every(b => b)) {
+              var ranges = data.map(d =>
+                d.getRange(d.elements.filter(el =>
+                  el.x >= focus.xl && el.x <= focus.xr
+                ))
+              );
+              var minYValue = Math.min(...ranges.map(r => r.yl).filter(y => !isNaN(y) && isFinite(y)));
+              var maxYValue = Math.max(...ranges.map(r => r.yr).filter(y => !isNaN(y) && isFinite(y)));
+              if(!isNaN(minYValue) && isFinite(minYValue) && !isNaN(maxYValue) && isFinite(maxYValue)) {
+                autoZoomSuccess = true;
+                focus.yl = minYValue;
```

Comment:
I think that we should add a check so that if the current base of the chart is 0 then we shouldn't change it

Reply:
Can you clarify what you mean by "current base"?

Reply:
The minimum value on the y-axis

Reply:
Ok, I'm not sure I understand. If the miniumum y-axis value (in the zoomed area) is 0, shouldn't it autoscale and set the lower bound to 0? I'm not seeing why it's a special case.

Reply:
I was thinking that the user might want to be able to for the minimum value on the y-axis to be 0 even if the smallest point is bigger than 0. And if they did so, then we wouldn't want to change the minimum value on the axis by zooming in that case. However, I'm not seeing that such an option exists right now, so we probably don't need to do anything after all. I think that the we add an option to manually set the minimum on the y-axis in the future then we should be sure to pay attention to that option here. But until we have such an option we probably don't need to do anything, so you can ignore my initial comment.